packages/cozy-client/src/withMutations.spec.jsx

```diff
@@ -14,6 +16,35 @@ describe('withMutations', () => {
 
   afterEach(() => {
     jest.resetAllMocks()
+    jest.restoreAllMocks()
+  })
+
+  it('should expose all additional mutations provided proptypes', () => {
+    const mutationsMock = client => ({
+      myMutation1: () => {},
+      myMutation2: () => {}
+    })
+    const ConnectedFoo = withMutations(mutationsMock)(() => <div />)
+    const wrapper = shallow(<ConnectedFoo />, {
+      context: { client: clientMock }
+    })
+    // mock for proptypes
+    jest.spyOn(console, 'error').mockImplementation(message => {
+      throw new Error(message)
+    })
```

Comment:
If we spyOn just before shallow rendering, prop checking will be done for free without the `       PropTypes.checkPropTypes(componentProps, 'prop', 'ConnectedFoo')` at the end ?

Reply:
It doesn't work for free :/ probably have to require again the `withMutations` after

Reply:
Actually, prop checking is not done at render time but at instantiation time.

Reply:
You're right, the good line to try/catch here is the withMutations() one, not the shallow, it's why my first tests didn't work I think